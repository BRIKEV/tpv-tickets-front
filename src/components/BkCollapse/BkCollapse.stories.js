import { storiesOf } from '@storybook/vue';

import BkCollapse from './BkCollapse.vue';

storiesOf('Components/BkCollapse', module)
  .add('default', () => ({
    components: { BkCollapse },
    data() {
      return {
        isOpened: false,
      };
    },
    methods: {
      handleToggle() {
        this.isOpened = !this.isOpened;
      },
    },
    template: `
      <BkCollapse :isOpened="isOpened">
        <template #header>
                <div>
                    <p>Collapse</p>
                    <i class="material-icons">
                    <button @click="handleToggle"></button>
                </div>
            </template>

        <div>
            <h2>Here will be the content</h2>
        </div>
      </BkCollapse>
        `,
  }));
