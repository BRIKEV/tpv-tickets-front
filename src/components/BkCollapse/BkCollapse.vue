<template>
  <div
    class="container"
    :class="{ 'not-expanded': !this.expanded }">
      <div
        data-cy="toggle-collapse"
        class="header"
        @click="toggleCardState">
          <p class="title" name="header">{{ title }}</p>
          <i class="material-icons icon">keyboard_arrow_down</i>
      </div>
      <transition
        name="fade"
        mode="out-in">
        <div
          class="content"
          v-if="expanded">
            <slot></slot>
        </div>
      </transition>
  </div>
</template>

<script>
export default {
  name: 'BkCollapse',
  props: {
    isOpened: {
      type: Boolean,
      value: false,
    },
    title: String,
  },
  data() {
    return {
      expanded: false,
    };
  },
  methods: {
    toggleCardState() {
      this.expanded = !this.expanded;
    },
  },
};
</script>

<style lang="scss" scoped>
@import "@/theme/index.scss";
$shadow: -1px 0px 7px -4px rgba(0,0,0,0.98);
  .fade-enter-active, .fade-leave-active {
    transition: opacity .5s;
  }
  .fade-enter, .fade-leave-to {
    opacity: 0;
  }
.container {
  position: relative;
  box-shadow: $shadow;
  padding: 15px;
  height: 100%;
  background: $white;
}
.header {
    display: flex;
  align-items: center;
  justify-content: space-between;
  cursor: pointer;
  icon: {
    transform: rotate(180deg);
  }
}
.container.not-expanded {
  .header {
    .icon {
      transform: rotate(-90deg);
    }
  }
}
.content {
  position: relative;
  margin-top: calculateRem(15px);

}
</style>
