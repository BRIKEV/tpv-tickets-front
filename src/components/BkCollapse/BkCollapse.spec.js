import { mount } from '@vue/test-utils';
import BkCollapse from '@/components/BkCollapse/BkCollapse.vue';
import localVue from '@/config/tests';

describe('BkCollapse component', () => {
  const defaultConfig = {
    localVue,
  };
  it('Should be render correctly', () => {
    const wrapper = mount(BkCollapse, defaultConfig);
    expect(wrapper.element).toMatchSnapshot();
  });
});
