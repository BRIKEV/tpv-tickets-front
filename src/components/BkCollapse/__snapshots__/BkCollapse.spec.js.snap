// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BkCollapse component Should be render correctly 1`] = `
<div
  class="container not-expanded"
>
  <div
    class="header"
  >
    <p
      class="title"
      name="header"
    >
      
    </p>
     
    <div
      class="arrow-container"
    >
      <!---->
       
      <i
        class="material-icons icon"
      >
        keyboard_arrow_down
      </i>
    </div>
  </div>
   
  <!---->
</div>
`;
